### PR TITLE
clean up sdks dir for each conformance test

### PR DESCRIPTION
**Pull request by tgummerer**
Multiple conformance tests can use the same provider.  When a provider gets used for a test we install all its dependencies.  Then in subsequent runs, when the provider is re-used those dependencies are already in the directory, and we complain that those newly installed dependencies don't exist in the original snapshot.

Fix this by cleaning up the sdks directory after each run, so we get a clean slate each time we codegen a provider.

We could potentially consider getting a list of providers when preparing the language tests, and just generating their code once, but that's a little bit more involved.

Fixes https://github.com/pulumi/pulumi/issues/15603

### CODE REVIEW

**Comment by Frassle** [comment]
hmm this should be a fix in the conformance test framework, not in the language test runners.
Also wonder if it should just skip the sdkgen if the sdk folder is already there. I think its safe to assume if the folder exists its because another test already wrote it.

**Comment by tgummerer** [comment]
Oh good idea, I didn't even think of that.  That should help the case here, and even better also help a bit with test performance!

File: cmd/pulumi-test-language/interface.go [hunk 1]
```diff
@@ -662,6 +662,17 @@ func (eng *languageTestServer) RunLanguageTest(
 
 		sdkName := fmt.Sprintf("%s-%s", pkg, version)
 		sdkTempDir := filepath.Join(token.TemporaryDirectory, "sdks", sdkName)
+		_, err = os.Stat(sdkTempDir)
+		if err == nil {
+			// If the directory already exists then we don't need to regenerate the SDK
+			sdkArtifact, err := languageClient.Pack(sdkTempDir, version, artifactsDir)
```
Comment (Frassle) [inline cmd/pulumi-test-language/interface.go:668]: I suppose ideally we wouldn't even re-pack it as it should be in the artifacts folder already, but this is probably fine for now.

**Review by Frassle — APPROVED** [review]

File: cmd/pulumi-test-language/interface.go (hunk 1 above)
Comment (tgummerer) [reply cmd/pulumi-test-language/interface.go:668]: Yeah, I think that would be better ideally.  But the artifact name is calculated in the fake language host, and we'll need some extra logic to make sure we can match that here.  I'd rather leave that until we have language support for all languages, then it's maybe easier to see the pattern there, or add additional methods in each language host.

**Pull request merged**